### PR TITLE
Xfs info documentation for master branch

### PR DESCRIPTION
**Pull request by PaulWay**
Improving the documentation and merging some features in the master branch from #575

### CODE REVIEW

File: insights/parsers/xfs_info.py [hunk 1]
```diff
@@ -104,54 +130,56 @@ def parse_content(self, content):
...
 
-        xfs_info = {}
         sect_info = None
         self.mount = ''
 
+        info_re = re.compile(r'^(?P<section>[\w-]+)?\s*' +
+                             r'=(?:(?P<specifier>\S+)(?:\s(?P<specval>\w+))?)?' +
+                             r'\s+(?P<keyvaldata>\w.*\w)$')
+        keyval_re = re.compile(r'(?P<key>[\w-]+)=(?P<value>\d+(?: blks)?)')
+
```
Comment (xiangce) [inline insights/parsers/xfs_info.py]: Line 136-139 are duplicated with line 127-131

File: insights/parsers/xfs_info.py [hunk 2]
```diff
@@ -20,60 +102,14 @@ class XFSInfo(Parser):
...
-        >>> data['meta-data']['agsize']  # blocks values not parsed
-        '16777184 blks'
-        >>> data['realtime']['rtextents']
-        0
     """
 
+    def __init__(self, context):
+        self.xfs_info = {}
+        self.data_size = 0
+        self.log_size = 0
```
Comment (xiangce) [inline insights/parsers/xfs_info.py]: How about shifting the initialization of `self.mount` and `self.device` from the parse_content to here too?

Comment (PaulWay) [reply insights/parsers/xfs_info.py]: A good point, might as well.

File: insights/parsers/xfs_info.py (hunk 1 above)
Comment (PaulWay) [reply insights/parsers/xfs_info.py]: Problems resolving merge conflicts - thanks for spotting that @xiangce

**Pull request merged**